druid/src/widget/image.rs

```diff
@@ -145,8 +145,17 @@ impl<T: Data> Widget<T> for Image {
     ) -> Size {
         bc.debug_check("Image");
 
-        if bc.is_width_bounded() {
-            bc.max()
+        // If either the width or height is constrained calculate a value so that the image fits
+        // in the size exactly. If it is unconstrained by both width and height take the size of
+        // the image.
+        if bc.is_width_bounded() && !bc.is_height_bounded() {
+            let max = bc.max();
```

Comment:
This doesn't have to be defined twice, it can be defined outside both `if` expressions just below the comment

Reply:
Sure, I didnt do it because I only do that when all branches need the variable and we only need it in the first 2.
I will wait for cmyr's review to fix these issues.